src/app/main/time-groups/edit-time-group/edit-time-group.component.ts

```diff
@@ -188,21 +191,35 @@ export class EditTimeGroupComponent implements OnInit {
     }
   }
   public nextpage() {
-    let nextPage = this.employeesPaginationData.next_page;
-    this.employessParams.page = nextPage;
-    if (nextPage) {
-      this.getEmployees()
+    if (this.employeesPaginationData.next_page) {
+      this.getEmployees("load", true)
     }
   }
 
-  public getEmployees() {
+  public getEmployees(type?: string, isLoadMore?: boolean) {
+    if(isLoadMore){
+      let nextPage = this.employeesPaginationData.next_page;
+      this.employessParams.page = nextPage;
+    }else{
+      this.employessParams.page = 1
+    }
+    if (type == "search") {
+      this.employeesList = []
+    }
     this.employeesLoading = true;
     this.employeesService.getEmployees(this.employessParams).subscribe((response: any) => {
       this.employeesList = this.employeesList.concat(response.employees)
+      this.employeesList.forEach(employee => {
+        employee.isInsideCurrentTimeGroup = employee.time_group?.name === this.timeGroup.name  ? true : false
+        if(this.timeGroup.employees && this.timeGroup.employees.length > 0){
+            employee.isInsideCurrentTimeGroup = this.timeGroup.employees.find(time_group_employee => time_group_employee.id == employee.id) ? true : false
```

Comment:
I think it's better to retrieve this value backend

Reply:
`timeGroup.employees` hold the time group persisted employees as well as the employees that are checked but still not saved. So if we filter the employees, the response will erase this selection. This is to check it again